operator/cmd/mesh/manifest-generate_test.go

```diff
@@ -95,6 +93,73 @@ func TestMain(m *testing.M) {
 	os.Exit(code)
 }
 
+func TestManifestGeneratePrometheus(t *testing.T) {
+	testDataDir = filepath.Join(operatorRootDir, "cmd/mesh/testdata/manifest-generate")
+	g := NewGomegaWithT(t)
+	_, objs, err := generateManifest("prometheus", "", liveCharts)
+	if err != nil {
+		t.Fatal(err)
+	}
+	want := []string{
+		"ClusterRole::prometheus-istio-system",
+		"ClusterRoleBinding::prometheus-istio-system",
+		"ConfigMap:istio-system:prometheus",
+		"Deployment:istio-system:prometheus",
+		"Service:istio-system:prometheus",
+		"ServiceAccount:istio-system:prometheus",
+	}
+	g.Expect(objectHashesOrdered(objs)).Should(ContainElements(want))
+}
+
+func TestManifestGenerateComponentHubTag(t *testing.T) {
+	testDataDir = filepath.Join(operatorRootDir, "cmd/mesh/testdata/manifest-generate")
+	g := NewGomegaWithT(t)
+	m, _, err := generateManifest("component_hub_tag", "", liveCharts)
+	if err != nil {
+		t.Fatal(err)
+	}
+	objs := parseObjectSetFromManifest(t, m)
+	g.Expect(objs.size()).Should(Not(Equal(0)))
+
+	tests := []struct {
+		deploymentName string
+		containerName  string
+		want           string
+	}{
+		{
+			deploymentName: "prometheus",
+			want:           "docker.io/prometheus:1.1.1",
+		},
+		{
+			deploymentName: "grafana",
+			want:           "grafana/grafana:6.5.2",
```

Comment:
we should override these so that we both get more coverage of the settings, and so that it doesn't change when we update these? Maybe you already have and set it to 6.5.2 explicitly though, in which case more of a nit that is could be more obviously a fake version

Reply:
It's set to that value in the base profile and I'm just testing here that it's not modified. It's a bit brittle if grafana version is updated but I think it will be trivial to fix if/when that happens. The alternative would be to pull the version out of the profile, which seems overkill. Otherwise this check could be removed completely.